pkgs/by-name/if/ifm-web/default.nix

```diff
@@ -0,0 +1,48 @@
+{
+  fetchurl,
+  stdenv,
+  lib,
+  php83,
+  writeShellScript,
+}:
+let
+  version = "4.0.2";
+  src = fetchurl {
+    url = "https://github.com/misterunknown/ifm/releases/download/v${version}/cdn.ifm.php";
+    hash = "sha256-37WbRM6D7JGmd//06zMhxMGIh8ioY8vRUmxX4OHgqBE=";
+  };
+
+  serve_script = writeShellScript "ifm-serve" ''
+    if [ $# -ne 3 ]; then
+      echo "Usage: $0 <listen address> <port> <data directory>";
+      exit 1;
+    fi
+
+    SERVE_DIR=$(dirname "$0")/../php/
+    IFM_ROOT_DIR="$3" ${lib.getExe php83} -S "$1:$2" -t "$SERVE_DIR"
+  '';
+
+in
+stdenv.mkDerivation {
+  pname = "ifm";
+  inherit version src;
+
+  phases = "installPhase";
+  installPhase = ''
+    mkdir -p $out/bin $out/php
+    ln -s $src $out/php/index.php
```

Comment:
We should copy the files to not retain a source reference and allow changes to the files.